### PR TITLE
Make deployment groups tokens backwards compatible

### PR DESCRIPTION
**Pull request by davidxia**

### CODE REVIEW

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java [hunk 1]
```diff
@@ -807,7 +807,9 @@ private RollingUpdateOp rollingUpdateDeploy(final ZooKeeperClient client,
                                                 deploymentGroup.getName());
 
     try {
-      return opFactory.nextTask(getDeployOperations(client, host, deployment, Job.EMPTY_TOKEN));
+      final String token = deploymentGroup.getRolloutOptions().getToken() == null ?
```
Comment (rohansingh) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: If you want, you could just do:

`final String token = firstNonNull(deploymentGroup.getRolloutOptions.getToken(), Job.EMPTY_TOKEN);`

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java [hunk 2]
```diff
@@ -832,17 +834,19 @@ private RollingUpdateOp rollingUpdateUndeploy(final ZooKeeperClient client,
...
         if (isSameJob && isOwnedByDeploymentGroup && deployment.getGoal().equals(Goal.START)) {
           // The job we want deployed is already deployed and set to run, so just leave it.
           continue;
         }
 
         try {
-          operations.addAll(getUndeployOperations(client, host, deployment.getJobId(),
-                                                  Job.EMPTY_TOKEN));
+          final String token =
+              rolloutOptions.getToken() == null ? Job.EMPTY_TOKEN : rolloutOptions.getToken();
```
Comment (rohansingh) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: Same as above, `firstNonNull` should work here and it's a bit easier to read.

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java (hunk 1 above)
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: Thanks

**Comment by davidxia** [comment]
@rohansingh staffan

**Comment by davidxia** [comment]
There were some spurious test failures. But it seems OK now.

**Comment by rohansingh** [comment]
:+1:

**Pull request merged**